subsys/shell/modules/kernel_service/thread/kill.c

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2025 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "kernel_shell.h"
+
+#include <kernel_internal.h>
+#include <zephyr/kernel.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+static int cmd_kernel_thread_kill(const struct shell *sh, size_t argc, char **argv)
+{
+	if (argc < 2) {
+		shell_error(sh, "Usage: thread kill <0xthread_id>");
+		return -EINVAL;
+	}
```

Comment:
This check seems unnecessary since you are already defining the minimum number of arguments in the command definition